crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -646,6 +655,33 @@ where
     }
 }
 
+pub struct MoveBundle<T> {
+    pub source: Entity,
+    pub target: Entity,
+    pub _bundle: PhantomData<T>,
+}
+
+impl<T> Command for MoveBundle<T>
+where
+    T: Bundle + 'static,
+{
+    fn write(self, world: &mut World) {
+        let bundle = if let Some(mut source) = world.get_entity_mut(self.source) {
+            source.remove_bundle::<T>()
+        } else {
+            panic!("Could not move a bundle (of type `{}`) for entity {:?} because it doesn't exist in this World.\n\
+                    If this command was added to a newly spawned entity, ensure that you have not despawned that entity within the same stage.\n\
+                    This may have occurred due to system order ambiguity, or if the spawning system has multiple command buffers", std::any::type_name::<T>(), self.source);
+        };
+
+        if let Some(bundle) = bundle {
+            if let Some(mut target) = world.get_entity_mut(self.target) {
```

Comment:
Should we panic if this fails?

Reply:
I think so, yes. I think either of the checks could fail

Obviously we want to work out a better system than panicking, but at the moment panicking has to do.